### PR TITLE
Open Toed Footwear

### PR DESCRIPTION
**Pull request by KasparoVy**
Feat. (feet) toeless jackboots.

:cl:
rscadd: Cutting open the toes of footwear so species with feet-claws can wear them.
rscadd: Toeless jackboots.
tweak: Cleans up glovesnipping and lighting a match with a shoe.
tweak: Lighting a match with a shoe produces a more appropriate message, handled at the same standard as glove-snippingboot-cutting.
/:cl:

Inspired by the toeless jackboots on Polaris/Bay. The toeless jackboot sprites are from Polaris/Bay.

This works the same as gloves. You take wirecutters, click the footwear, and bam! Changes the sprites, and opens the toe: Letting Unathi and Tajara put on the footwear if they weren't allowed to already.

If you try to cut open the toes of already toe-cut footwear, it won't let you.
If you try to cut open the toes of footwear that don't support that functionality, it does nothing.

*Possible Messages.*
opentoedbootsmessages

*Sprites.*
opentoedbootsicons
*The white on the feet of the Unathi is the toes sticking out.*
*If you can't see the toes on the Tajara, the shape of the foot appears different as there isn't a bulky boot-toe on the end.*

### CODE REVIEW

**Comment by IcyV** [comment]
I was going to make exactly this soon. You are my favorite person right now Kasparov.

**Comment by Fox-McCloud** [comment]
There shouldn't be two attackby()'s.

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -330,6 +332,26 @@ BLIND     // can't see anything
...
+					desc = "[desc] They have had their toes opened up."
+					species_restricted -= "Unathi"
+					species_restricted -= "Tajaran"
+					icon_state = "[icon_state]_opentoe"
+					item_state = "[item_state]_opentoe"
+			else if(cut_open == 1)
+				user << "<span class='notice'>[src] have already had their toes cut open!</span>"
+				update_icon()
+			return
+			..()
```
Comment (Fox-McCloud) [inline code/modules/clothing/clothing.dm]: I don't even; this can't even run.

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -330,6 +332,26 @@ BLIND     // can't see anything
 		"Vox" = 'icons/mob/species/vox/shoes.dmi'
 		)
 
+/obj/item/clothing/shoes/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/wirecutters))
+		if(can_cut_open == 1)
+			if(cut_open == 0)
```
Comment (Fox-McCloud) [inline code/modules/clothing/clothing.dm]: explicitly checking for 0 or 1 when it's only 0 or 1 is unnecessary.

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -330,6 +332,26 @@ BLIND     // can't see anything
 		"Vox" = 'icons/mob/species/vox/shoes.dmi'
 		)
 
+/obj/item/clothing/shoes/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/wirecutters))
+		if(can_cut_open == 1)
```
Comment (Fox-McCloud) [inline code/modules/clothing/clothing.dm]: if(can_cut_open) is sufficient.

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -330,6 +332,26 @@ BLIND     // can't see anything
 		"Vox" = 'icons/mob/species/vox/shoes.dmi'
 		)
 
+/obj/item/clothing/shoes/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/wirecutters))
+		if(can_cut_open == 1)
+			if(cut_open == 0)
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 100, 1)
+				user.visible_message("\red [user] cuts open the toes of [src].","\red You cut open the toes of [src].")
```
Comment (Fox-McCloud) [inline code/modules/clothing/clothing.dm]: Spans

File: code/modules/clothing/clothing.dm [hunk 5]
```diff
@@ -330,6 +332,26 @@ BLIND     // can't see anything
...
+				cut_open = 1
+				if("exclude" in species_restricted)
+					name = "mangled [name]"
+					desc = "[desc] They have had their toes opened up."
+					species_restricted -= "Unathi"
+					species_restricted -= "Tajaran"
+					icon_state = "[icon_state]_opentoe"
+					item_state = "[item_state]_opentoe"
+			else if(cut_open == 1)
```
Comment (Fox-McCloud) [inline code/modules/clothing/clothing.dm]: I don't think this needs to be an else if....also, why does it update the icon when it's already cut open?

**Comment by Fox-McCloud** [comment]
Random fact (For those who didn't know); there are, in fact, pairs of jack sandals in sec.

**Comment by KasparoVy** [comment]
I'll sort this in 6 hours, at work right now

Thanks for pointing out the particulars, I'll address each of them when I'm back

Figured I'd put it up for review before I had a nap

**Comment by NTSAM** [comment]
Jacksandals are stupid.

**Comment by Spacemanspark** [comment]
I like both this and the jacksandals.
Just add this and keep the sandals.

**Comment by Plotron** [comment]
@Spacemanspark got it right.

**Comment by KasparoVy** [comment]
This doesn't remove anything, this just adds the ability to mangle boots and whatever else for CLAWFOOTBEASTS :B 

Just a few more hours and I'll be back on this.

File: code/modules/clothing/clothing.dm (hunk 5 above)
Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: No idea why I put the icon updates down there, they should be just below cut_open=1 and above the if("exclude"...

File: code/modules/clothing/clothing.dm (hunk 1 above)
Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: What exactly are you referring to?

File: code/modules/clothing/clothing.dm (hunk 3 above)
Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: Thanks, I was iffy on that

File: code/modules/clothing/clothing.dm (hunk 2 above)
Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: Thanks, I was iffy on that. I'll make the change

File: code/modules/clothing/clothing.dm (hunk 4 above)
Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: I'll change glovecutting to use spans while I'm at it.

File: code/modules/clothing/clothing.dm (hunk 1 above)
Comment (tigercat2000) [inline code/modules/clothing/clothing.dm]: The `..()` call, it's under `return`- thus it can never run

Comment (Crazylemon64) [inline code/modules/clothing/clothing.dm]: It's a parent call after a return - return aborts the function completely, so the parent call won't get reached at all

Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: Huh. I referred to how this was done in glove snipping. Guess there's some tidying up to do. Thanks for clearing that up!

**Comment by KasparoVy** [comment]
Had to stay an extra hour. Anyway, that should sort it all. The messages will no longer be in red, but they weren't that important anyway.

Noticeable yet more unobtrusive blue seems a better fit, anyhow.

File: code/modules/clothing/clothing.dm [hunk 6]
```diff
@@ -178,19 +178,20 @@ BLIND     // can't see anything
 
 /obj/item/clothing/gloves/attackby(obj/item/weapon/W, mob/user, params)
 	if(istype(W, /obj/item/weapon/wirecutters))
-		if(clipped == 0)
+		if(!clipped)
 			playsound(src.loc, 'sound/items/Wirecutter.ogg', 100, 1)
-			user.visible_message("\red [user] snips the fingertips off [src].","\red You snip the fingertips off [src].")
+			user.visible_message("<span class='notice'>[user] snips the fingertips off [src].</span>","<span class='notice'>You snip the fingertips off [src].</span>")
```
Comment (tigercat2000) [inline code/modules/clothing/clothing.dm]: you could just use `<span class='warning'>`

Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: Sweet, that's perfect. Thank you. Commit incoming.

**Comment by sasanek12** [comment]
:+1::+1::+1:
Finally something else than sandals

**Comment by Iamgoofball** [comment]
i didnt know Quentin Tarantino coded for SS13

File: code/modules/clothing/clothing.dm [hunk 7]
```diff
@@ -178,19 +178,20 @@ BLIND     // can't see anything
...
 			clipped = 1
+			name = "mangled [name]"
+			desc = "[desc] They have had the fingertips cut off of them."
 			if("exclude" in species_restricted)
-				name = "mangled [name]"
-				desc = "[desc] They have had the fingertips cut off of them."
 				species_restricted -= "Unathi"
 				species_restricted -= "Tajaran"
+			update_icon()
 		else if(clipped == 1)
```
Comment (Fox-McCloud) [inline code/modules/clothing/clothing.dm]: else

File: code/modules/clothing/clothing.dm [hunk 8]
```diff
@@ -333,16 +336,34 @@ BLIND     // can't see anything
...
-		if(M.lit == 0)
+		if(!M.lit)
 			M.lit = 1
 			M.icon_state = "match_lit"
 			processing_objects.Add(M)
 			M.update_icon()
-			user << "<span class='notice'>You strike the [M] along your [src] to light it.</span>"
-		else if(M.lit == 1)
+			user << "<span class='warning'>You strike the [M] along [src] to light it.</span>"
+		else if(M.lit)
```
Comment (Fox-McCloud) [inline code/modules/clothing/clothing.dm]: else

**Comment by Fox-McCloud** [comment]
Much better!

File: code/modules/clothing/clothing.dm (hunk 8 above)
Comment (FalseIncarnate) [inline code/modules/clothing/clothing.dm]: No, leave as-is for this.

Burnt matches are lit = -1, so we have to check for lit ==1 for burning ones.

Comment (Fox-McCloud) [inline code/modules/clothing/clothing.dm]: ugh, yeah, leave it as is then.

Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: Sounds good, brbcommit

**Comment by Iamgoofball** [comment]
Maybe instead of adding creepy foot fetish stuff, you could maybe make races not have arbitrary and retardedly stupid restrictions on what they can and can't wear?

If the sprites dont fit it then consider a resprite to make it less fetishy.

**Comment by NTSAM** [comment]
sandals are fetishy???

I wear sandals all the time so i guess i'm a perv

**Comment by KasparoVy** [comment]
I was thinking of a way in which to implement the open-toed boot sprite along that line of thinking and this seemed like the better way to go about it.
Adding yet another item to clutter up security lockers and vendors and the like didn't seem like as good an idea.

This works with restrictions that make sense, and if someone puts in some more footclaw-accommodating sprites later on, all it'll take is one more line of code

Else, the footwear will look like clown's shoes due to the expanded toe. *shrug*

File: code/modules/clothing/clothing.dm (hunk 7 above)
Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: Dealing with this now, going to squash commits after this one then we should be good
Thanks for pointing that out

**Pull request merged**